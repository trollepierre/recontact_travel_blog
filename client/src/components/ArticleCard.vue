<template>
  <div class="article-card">
    <article class="article">
      <header class="article__header">
        <a :href="articleUrl" @click.prevent.once="viewArticle">
          <h2 class="article__title">{{ articleTitle }}</h2>
        </a>
      </header>
      <div class="article__content">
        <img class="article__image" :src="article.imgLink" width="200"/>
      </div>
      <footer class="article__footer">
        <template v-if="adminMode">
          <button class="article__update-button" :disabled="isUpdateClicked"
                  @click.prevent.once="updateArticle">
            {{ $t("repairArticle") }}
          </button>
        </template>
        <template v-else>
          <button class="article__view-button"
                  @click.prevent.once="viewArticle">
            {{ $t("goToArticle") }}
          </button>
          <a :href="article.galleryLink" target="_blank" class="article__dropbox">
            <button class="article__dropbox-button">
              {{ $t("viewGallery") }}
            </button>
          </a>
        </template>
      </footer>
    </article>
  </div>
</template>

<script>
  import articlesApi from '@/api/articles';
  import notificationsService from '@/services/notifications';
  import translationsService from '@/services/translations';

  export default {
    name: 'ArticleCard',
    props: ['article', 'adminMode'],
    data() {
      return {
        isUpdateClicked: false,
      };
    },
    computed: {
      articleUrl() {
        return `/articles/${this.article.dropboxId}`;
      },
      articleTitle() {
        return translationsService.getTitle(this.article);
      },
    },
    methods: {
      viewArticle() {
        this.goToArticle();
      },

      updateArticle() {
        this.disableDeleteButton();
        notificationsService.information(this, this.$t('syncLaunched'));
        articlesApi.update(this.article.dropboxId)
          .then(() => {
            notificationsService.removeInformation(this);
            notificationsService.success(this, this.$t('syncDone'));
          })
          .then(() => this.goToArticle())
          .catch((err) => {
            notificationsService.removeInformation(this);
            notificationsService.error(this, `${this.$t('syncError')} ${err}`);
          });
      },

      disableDeleteButton() {
        this.isUpdateClicked = true;
      },

      goToArticle() {
        this.$router.push(this.articleUrl);
      },
    },

    i18n: {
      messages: {
        fr: {
          repairArticle: 'Réparer l‘article',
          goToArticle: 'Voir l‘article',
          viewGallery: 'Voir les photos',
          syncLaunched: 'La synchronisation est lancée ! Patientez quelques secondes...',
          syncDone: 'La synchronisation s‘est effectuée sans problème !',
          syncError: 'Erreur : Problème durant la synchronisation :',
        },
        en: {
          repairArticle: 'Repair the article',
          goToArticle: 'Read the article',
          viewGallery: 'Discover the pictures',
          syncLaunched: 'The synchronisation is launched! Please wait...',
          syncDone: 'The synchronisation succeeds!',
          syncError: 'Error during the synchronisation:',
        },
      },
    },
  };
</script>

<style scoped>
  h2 {
    font-weight: normal;
  }

  img {
    max-width: 100%;
    max-height: 100%;
  }

  .sr-only {
    position: absolute;
    width: 1px;
    height: 1px;
    padding: 0;
    margin: -1px;
    overflow: hidden;
    clip: rect(0, 0, 0, 0);
    border: 0;
    display: block;
  }

  .article {
    min-width: 260px;
    max-width: 260px;
    background: #ffffff;
    border-radius: 4px !important;
    box-shadow: 0 1px 1px rgba(0, 0, 0, .15);
    border: 1px solid rgba(0, 0, 0, .09);
    display: flex;
    flex-direction: column;
    color: #535a60;
  }

  .article__header {
    border-bottom: 1px solid #e6e6e6;
    height: 34px;
    padding: 15px;
    display: flex;
    justify-content: space-between;
  }

  .article__header a {
    margin: auto;
  }

  .article__title {
    font-size: 16px;
    font-weight: 700;
    line-height: 17px;
    color: #07c;
    margin: auto;
    overflow: hidden;
    text-overflow: ellipsis;
    max-width: 200px;
  }

  .article__content {
    font-size: 15px;
    padding: 15px;
    height: 150px;
    color: #000;
    text-align: center;
    display: flex;
    align-content: center;
    justify-content: center;
  }

  .article__content > p {
    margin-top: 0;
  }

  .article__footer {
    text-align: center;
    padding: 15px;
    border-top: 1px solid #e6e6e6;
  }

  .article__footer button {
    text-transform: uppercase;
    color: #d14800;
    background: #ffffff;
    border: 1px solid #d14800;
    cursor: pointer;
    padding: 15px 30px;
    border-radius: 4px;
    width: 100%;
    margin-bottom: 10px;
    font-weight: 700;
  }

  .article__footer button:hover {
    background: #d14800;
    color: #ffffff;
  }

  .article__footer button:disabled,
  .article__footer button:active {
    background: #BDBDBD;
    border-color: #616161;
    color: #FAFAFA;
    cursor: auto;
  }

  @media only screen and (max-width: 640px) {
    .article {
      max-width: inherit;
    }
  }
</style>
