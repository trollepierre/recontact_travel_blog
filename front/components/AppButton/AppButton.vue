<template>
  <component
    :is="tag"
    type="button"
    class="button"
    :to="to"
    @click="(e) => allowMultipleClick ? onClick(e) : undefined"
    @click.prevent.once="(e) => allowMultipleClick ? undefined : onClick(e)">
    {{ text }}
  </component>
</template>
<script>
  export default {
    name: 'AppButton',
    props: {
      text: { type: String, default: () => '' },
      isLink: { type: Boolean, default: () => false },
      to: { type: String, default: () => undefined },
      tag: { type: String, default: () => 'button' },
      allowMultipleClick: { type: Boolean, default: () => false },
    },
    methods: {
      onClick(e) {
        e.preventDefault()
        this.$emit('click')
      },
    },
  }
</script>

<style lang="scss" scoped>
.button {
  line-height: 28px;
  color: $button-color;
  text-decoration: unset;
  font-size: 11px;
  font-family: serif;
  text-transform: uppercase;
  background: $button-bg;
  border: 1px solid $button-color;
  cursor: pointer;
  padding: 3px 5px;
  border-radius: 4px;
  width: 100%;
  font-weight: 700;
}

.button:hover {
  background: $button-hover;
  color: $button-bg;
}

</style>
