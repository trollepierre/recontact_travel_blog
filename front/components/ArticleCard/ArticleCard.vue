<template>
  <div class="article-card">
    <article class="article">
      <header class="article__header">
        <a
          :href="articleUrl"
          @click.prevent.once="viewArticle">
          <h2 class="article__title">
            {{ articleTitle }}
          </h2>
        </a>
      </header>
      <div
        class="article__content"
        @click.prevent.once="viewArticle">
        <img
          :src="article.imgLink"
          :alt="articleTitle"
          class="article__image"
          width="200">
      </div>
      <footer class="article__footer">
        <template v-if="adminMode">
          <button
            :disabled="isUpdateClicked"
            class="article__update-button"
            @click.prevent.once="updateArticle">
            {{ $t("repairArticle") }}
          </button>
          <button
            :disabled="isDeleteClicked"
            class="article__delete-button article__footer_hidden"
            @click.prevent.once="deleteArticle">
            {{ $t("deleteArticle") }}
          </button>
        </template>
        <template v-else>
          <button
            class="article__view-button"
            @click.prevent.once="viewArticle">
            {{ $t("goToArticle") }}
          </button>
          <a
            :href="article.galleryLink"
            target="_blank"
            class="article__dropbox">
            <button class="article__dropbox-button">
              {{ $t("viewGallery") }}
            </button>
          </a>
        </template>
      </footer>
    </article>
  </div>
</template>

<script>
  import articlesApi from '../../services/api/articles'
  import notificationsService from '../../services/services/notifications'
  import translationsService from '../../services/services/translations'

  export default {
    name: 'ArticleCard',
    props: {
      adminMode: { type: Boolean, default: () => false },
      article: { type: Object, default: () => {} },
    },
    data() {
      return {
        isUpdateClicked: false,
        isDeleteClicked: false,
      }
    },
    computed: {
      articleUrl() {
        return `/articles/${this.article.dropboxId}`
      },
      articleTitle() {
        const language = this.$store.state.locale
        return translationsService.getTitle(this.article, language)
      },
    },
    methods: {
      viewArticle() {
        this.goToArticle()
      },

      updateArticle() {
        this.trackEvent()
        this.disableUpdateButton()
        notificationsService.information(this, this.$t('syncLaunched'))
        articlesApi.update(this.article.dropboxId)
          .then(() => {
            notificationsService.removeInformation(this)
            notificationsService.success(this, this.$t('syncDone'))
          })
          .then(() => this.goToArticle())
          .catch(err => {
            notificationsService.removeInformation(this)
            notificationsService.error(this, `${this.$t('syncError')} ${err}`)
          })
      },

      disableUpdateButton() {
        this.isUpdateClicked = true
      },

      trackEvent() {
        this.$ga.event({
          eventCategory: 'Article Card',
          eventAction: 'update',
          eventLabel: `article ${this.article.dropboxId} is updated`,
        })
      },

      deleteArticle() {
        this.disableDeleteButton()
        notificationsService.information(this, this.$t('deleteLaunched'))
        articlesApi.delete(this.article.dropboxId)
          .then(() => {
            notificationsService.removeInformation(this)
            notificationsService.success(this, this.$t('deleteDone'))
          })
          .catch(err => {
            notificationsService.removeInformation(this)
            notificationsService.error(this, `${this.$t('deleteError')} ${err}`)
          })
      },

      disableDeleteButton() {
        this.isDeleteClicked = true
      },

      goToArticle() {
        this.$router.push(this.articleUrl)
      },
    },

    i18n: {
      messages: {
        fr: {
          repairArticle: 'Réparer l’article',
          deleteArticle: 'Supprimer l’article',
          goToArticle: 'Voir l’article',
          viewGallery: 'Voir les photos',
          syncLaunched: 'La synchronisation est lancée ! Patientez quelques secondes...',
          syncDone: 'La synchronisation s’est effectuée sans problème  !',
          syncError: 'Erreur : Problème durant la synchronisation  :',
          deleteLaunched: 'La suppression est lancée ! Patientez quelques secondes...',
          deleteDone: 'La suppression s’est effectuée sans problème  !',
          deleteError: 'Erreur : Problème durant la suppression  :',
        },
        en: {
          repairArticle: 'Repair the article',
          deleteArticle: 'Delete the article',
          goToArticle: 'Read the article',
          viewGallery: 'Discover the pictures',
          syncLaunched: 'The synchronisation is launched! Please wait...',
          syncDone: 'The synchronisation succeeds!',
          syncError: 'Error during the synchronisation:',
          deleteLaunched: 'The deletion is launched! Please wait...',
          deleteDone: 'The deletion succeeds!',
          deleteError: 'Error during the deletion:',
        },
      },
    },
  }
</script>

<style scoped>
  h2 {
    font-weight: normal;
  }

  img.article__image {
    max-width: 100%;
    max-height: 40vw;
  }

  @media only screen and (max-width: 640px) {
    img.article__image {
      max-height: 100%;
    }
  }

  .sr-only {
    position: absolute;
    width: 1px;
    height: 1px;
    padding: 0;
    margin: -1px;
    overflow: hidden;
    clip: rect(0, 0, 0, 0);
    border: 0;
    display: block;
  }

  .article {
    min-width: 260px;
    max-width: 260px;
    background: #ffffff;
    border-radius: 4px !important;
    box-shadow: 0 1px 1px rgba(0, 0, 0, .15);
    border: 1px solid rgba(0, 0, 0, .09);
    display: flex;
    flex-direction: column;
    color: #535a60;
  }

  .article__header {
    border-bottom: 1px solid #e6e6e6;
    height: 34px;
    padding: 15px;
    display: flex;
    justify-content: space-between;
  }

  .article__header a {
    margin: auto;
  }

  .article__title {
    font-size: 16px;
    font-weight: 700;
    line-height: 17px;
    color: #07c;
    margin: auto;
    overflow: hidden;
    text-overflow: ellipsis;
    max-width: 200px;
  }

  .article__content {
    font-size: 15px;
    padding: 15px;
    height: 150px;
    color: #000;
    text-align: center;
    display: flex;
    align-content: center;
    justify-content: center;
  }

  .article__content > p {
    margin-top: 0;
  }

  .article__footer {
    text-align: center;
    padding: 15px;
    border-top: 1px solid #e6e6e6;
  }

  .article__footer button {
    text-transform: uppercase;
    color: #d14800;
    background: #ffffff;
    border: 1px solid #d14800;
    cursor: pointer;
    padding: 15px 30px;
    border-radius: 4px;
    width: 100%;
    margin-bottom: 10px;
    font-weight: 700;
  }

  .article__footer button.article__footer_hidden {
    color: #f7b5a9;
  }

  .article__footer button:hover {
    background: #d14800;
    color: #ffffff;
  }

  .article__footer button:disabled,
  .article__footer button:active {
    background: #BDBDBD;
    border-color: #616161;
    color: #FAFAFA;
    cursor: auto;
  }

  @media only screen and (max-width: 640px) {
    .article {
      max-width: inherit;
    }
  }
</style>
