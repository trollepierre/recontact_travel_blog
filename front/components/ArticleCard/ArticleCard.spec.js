import Vue from 'vue'
import VueLazyload from 'vue-lazyload'
import VueI18n from 'vue-i18n'
import Vuex from 'vuex' // eslint-disable-line import/no-extraneous-dependencies
import VueRouter from 'vue-router' // eslint-disable-line import/no-extraneous-dependencies

import AppButton from '@/components/AppButton/AppButton'
import ArticleCard from './ArticleCard.vue'
import articlesApi from '../../services/api/articles'
import chaptersApi from '../../services/api/chapters'
import notificationsService from '../../services/services/notifications'
import translationsService from '../../services/services/translations'

describe('Component | ArticleCard.vue', () => {
  let localVue
  let store
  let wrapper
  const galleryLink = 'https://www.dropbox.com/sh/k79oskpopi9lm8v/AABst0JslmKYw3Rhx9BjwJxMa?dl=0'
  let article
  let propsData
  const dropboxId = '58'
  const router = {
    init: jest.fn(),
    push: jest.fn(),
    history: {},
  }

  beforeEach(() => {
    localVue = createLocalVue()
    localVue.use(VueI18n)
    localVue.use(VueLazyload)
    localVue.use(VueRouter)
    localVue.use(Vuex)

    article = {
      dropboxId,
      enTitle: 'Pierre in Costa Rica',
      frTitle: 'Pierre au Costa Rica',
      imgLink: 'webf',
      galleryLink,
    }
    propsData = {
      article,
    }
    store = new Vuex.Store({ actions: {}, state: { locale: 'en' } })
    wrapper = shallowMount(ArticleCard, {
      localVue, propsData, router, store,
    })
  })

  describe('when adminMode is not defined', () => {
    beforeEach(() => {
      translationsService.getTitle = jest.fn()
      translationsService.getTitle.mockReturnValue('Pierre somewhere')
      wrapper = shallowMount(ArticleCard, {
        localVue, propsData, router, store,
      })
    })

    describe('template', () => {
      it('should match snapshot', () => {
        expect(wrapper).toMatchSnapshot()
      })

      it('should match snapshot when lazy is false', () => {
        propsData = {
          article,
          lazy: false,
        }
        wrapper = shallowMount(ArticleCard, {
          localVue, propsData, router, store,
        })
        expect(wrapper).toMatchSnapshot()
      })

      it('should have enabled article button', () => {
        expect(wrapper.find('.article__view-button').disabled).toBeUndefined()
      })

      it('should have enabled dropbox button', () => {
        expect(wrapper.find('.article__dropbox-button').disabled).toBeUndefined()
      })
    })

    describe('$data', () => {
      it('should have isUpdateClicked property set to false', () => {
        expect(wrapper.vm.isUpdateClicked).toEqual(false)
      })
      it('should have isDeleteClicked property set to false', () => {
        expect(wrapper.vm.isDeleteClicked).toEqual(false)
      })
    })

    describe('computed', () => {
      describe('#articleUrl', () => {
        it('should return /articles/:id', () => {
          expect(wrapper.vm.articleUrl).toEqual('/articles/58')
        })
      })

      describe('#articleTitle', () => {
        it('should return articleName', () => {
          expect(wrapper.vm.articleTitle).toEqual('Pierre somewhere')
        })
      })
    })

    describe('methods', () => {
      describe('#disableUpdateButton', () => {
        it('should set isUpdateClicked to true', () => {
          wrapper.vm.disableUpdateButton()

          expect(wrapper.vm.isUpdateClicked).toEqual(true)
        })
      })

      describe('#viewArticle', () => {
        it('should redirect to /articles/:idArticle', () => {
          wrapper = shallowMount(ArticleCard, {
            localVue, propsData, router, store,
          })

          wrapper.vm.viewArticle()

          expect(router.push).toHaveBeenCalledWith('/articles/58')
        })
      })

      describe('#goToArticle', () => {
        it('should redirect to /articles/:idArticle', () => {
          wrapper = shallowMount(ArticleCard, {
            localVue, propsData, router, store,
          })

          wrapper.vm.goToArticle()

          expect(router.push).toHaveBeenCalledWith('/articles/58')
        })
      })

      describe('#updateArticle', () => {
        beforeEach(() => {
          articlesApi.update = jest.fn()
          notificationsService.information = jest.fn()
          notificationsService.warn = jest.fn()

          notificationsService.error = jest.fn()
        })

        it('should set isUpdateClicked to true', () => {
          articlesApi.update.mockResolvedValue({})

          wrapper.vm.updateArticle()

          expect(wrapper.vm.isUpdateClicked).toEqual(true)
        })

        it('should call delete article api', () => {
          articlesApi.update.mockResolvedValue({})

          wrapper.vm.updateArticle()

          expect(articlesApi.update).toHaveBeenCalledWith(dropboxId)
        })

        it('should display success toast notification before synchronisation calls', () => {
          articlesApi.update.mockResolvedValue({})

          wrapper.vm.updateArticle()

          const message = 'The synchronisation is launched! Please wait...'
          expect(notificationsService.information).toHaveBeenCalledWith(message)
        })

        it('should redirect to /article/id', () => {
          localVue = createLocalVue()
          localVue.use(VueI18n)
          localVue.use(VueRouter)
          localVue.use(Vuex)
          const consoleError = console.error
          console.error = jest.fn()
          articlesApi.update.mockResolvedValue({})
          wrapper = shallowMount(ArticleCard, {
            localVue, propsData, router, store,
          })

          wrapper.vm.updateArticle()

          return Vue.nextTick().then(() => {
            expect(router.push).toHaveBeenCalledWith('/articles/58')

            console.error = consoleError
          })
        })

        it('should display success toast notification when synchronisation succeeds', () => {
          articlesApi.update.mockResolvedValue({})

          wrapper.vm.updateArticle()

          return Vue.nextTick().then(() => {
            const message = 'The synchronisation succeeds!'
            expect(notificationsService.information).toHaveBeenCalledWith(message)
          })
        })

        // xit('should display error toast notification when synchronisation fails', () => {
        //   articlesApi.update.mockRejectedValue(new Error('Expected error'))
        //
        //   wrapper.vm.updateArticle()
        //
        //   return Vue.nextTick().then(() => {
        //
        //     const message = 'syncError Error: Expected error'
        //     expect(notificationsService.error).toHaveBeenCalledWith(message)
        //   })
        // })
      })
    })

    describe('events', () => {
      describe('clicking on button "Voir l\'article"', () => {
        it('should redirect to /article/id', () => {
          wrapper.findAllComponents(AppButton).at(0).vm.$emit('click')

          expect(router.push).toHaveBeenCalledWith('/articles/58')
        })
      })

      describe('clicking on title', () => {
        it('should redirect to /article/id', () => {
          wrapper.find('.article__link').trigger('click')

          expect(router.push).toHaveBeenCalledWith('/articles/58')
        })
      })

      describe('clicking on image', () => {
        it('should redirect to /article/id', () => {
          wrapper.find('.article__content').trigger('click')

          expect(router.push).toHaveBeenCalledWith('/articles/58')
        })
      })
    })
  })

  describe('when adminMode is true', () => {
    beforeEach(() => {
      translationsService.getTitle = jest.fn()
      translationsService.getTitle.mockReturnValue('Pierre somewhere')
      propsData = {
        article,
        adminMode: true,
      }
      wrapper = shallowMount(ArticleCard, {
        localVue, propsData, router, store,
      })
    })

    it('should match snapshot', () => {
      expect(wrapper).toMatchSnapshot()
    })

    describe('clicking on button "repare the article"', () => {
      beforeEach(() => {
        articlesApi.update = jest.fn()
        articlesApi.update.mockResolvedValue({})
        notificationsService.information = jest.fn()
        notificationsService.warn = jest.fn()

        notificationsService.error = jest.fn()
      })

      it('should call articlesApi', () => {
        wrapper.findAllComponents(AppButton).at(1).vm.$emit('click')

        expect(articlesApi.update).toHaveBeenCalledWith('58')
        expect(wrapper.vm.isUpdateClicked).toEqual(true)
      })
    })

    describe('clicking on button "repare the chapter"', () => {
      beforeEach(() => {
        chaptersApi.update = jest.fn()
        chaptersApi.update.mockResolvedValue({})
        notificationsService.information = jest.fn()
        notificationsService.warn = jest.fn()

        notificationsService.error = jest.fn()
      })

      it('should not call chaptersApi when input unchanged', () => {
        console.log = jest.fn()

        wrapper.findAllComponents(AppButton).at(0).vm.$emit('click')

        expect(notificationsService.error).toHaveBeenCalledOnceWith('incorrect chapter number: 0')
        expect(chaptersApi.update).not.toHaveBeenCalled()
      })

      it('should call chaptersApi when input changed', () => {
        wrapper.setData({ chapterToRepair: '2' })
        wrapper.findAllComponents(AppButton).at(0).vm.$emit('click')

        expect(chaptersApi.update).toHaveBeenCalledWith('58', 2)
        expect(wrapper.vm.isUpdateChapterClicked).toEqual(true)
      })
    })
  })

  describe('locales', () => {
    const languages = Object.keys(ArticleCard.i18n.messages)

    it('contains 2 languages', () => {
      expect(languages).toHaveLength(2)
      expect(languages).toEqual(['fr', 'en'])
    })

    describe('each language', () => {
      describe('fr', () => {
        const locales = Object.keys(ArticleCard.i18n.messages.fr)

        it('contains 11 locales', () => {
          expect(locales).toHaveLength(11)
          expect(locales).toEqual([
            'repairArticle',
            'repairChapter',
            'deleteArticle',
            'goToArticle',
            'viewGallery',
            'syncLaunched',
            'syncDone',
            'syncError',
            'deleteLaunched',
            'deleteDone',
            'deleteError',
          ])
        })
      })

      describe('en', () => {
        const locales = Object.keys(ArticleCard.i18n.messages.en)

        it('contains 11 locales', () => {
          expect(locales).toHaveLength(11)
          expect(locales).toEqual([
            'repairArticle',
            'repairChapter',
            'deleteArticle',
            'goToArticle',
            'viewGallery',
            'syncLaunched',
            'syncDone',
            'syncError',
            'deleteLaunched',
            'deleteDone',
            'deleteError',
          ])
        })
      })
    })
  })
})
