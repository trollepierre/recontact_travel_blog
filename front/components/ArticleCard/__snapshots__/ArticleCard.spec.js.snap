// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component | ArticleCard.vue when adminMode is not defined template should match snapshot 1`] = `
<article class="article"><a href="/articles/58" class="article__link">
    <h2 class="article__title">
      Pierre somewhere
    </h2>
  </a>
  <div class="article__content"><img alt="Pierre somewhere" rel="noreferrer" class="article__image" data-src="webf" src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7" lazy="loading"></div>
  <footer class="article__footer"><button class="article__footer__button">
      Read the article
    </button> <a href="https://www.dropbox.com/sh/k79oskpopi9lm8v/AABst0JslmKYw3Rhx9BjwJxMa?dl=0" target="_blank" rel="noreferrer" class="article__dropbox article__footer__button">
      Discover the pictures
    </a></footer>
</article>
`;

exports[`Component | ArticleCard.vue when adminMode is not defined template should match snapshot when lazy is false 1`] = `
<article class="article"><a href="/articles/58" class="article__link">
    <h2 class="article__title">
      Pierre somewhere
    </h2>
  </a>
  <div class="article__content"><img src="webf" alt="Pierre somewhere" width="200" class="article__image important"></div>
  <footer class="article__footer"><button class="article__footer__button">
      Read the article
    </button> <a href="https://www.dropbox.com/sh/k79oskpopi9lm8v/AABst0JslmKYw3Rhx9BjwJxMa?dl=0" target="_blank" rel="noreferrer" class="article__dropbox article__footer__button">
      Discover the pictures
    </a></footer>
</article>
`;
